### PR TITLE
feat: add jsx-runtime

### PR DESCRIPTION
**Pull request by cqh963852**
Add jsx-runtime package to addons

### CODE REVIEW

File: addons/van_jsx/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      name: string;
+    }
+
+    const Hello = (props: IProps) => {
+      const { name } = props;
+      return <div>Hello {name}</div>;
+    };
+
+    export default Hello;
+    ```
```
Comment (Tao-VanJS) [inline addons/van_jsx/README.md:93]: Are you able to provide a preview link for the sample app (for instance with the help of CodeSandbox)?

File: addons/van_jsx/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      "import": "./src/jsx-dev-runtime.js",
+      "types": "./src/jsx-dev-runtime.d.ts"
+    }
+  },
+  "files": [
+    "src",
+    "jsx-runtime.d.ts"
+  ],
+  "scripts": {},
+  "author": "",
```
Comment (Tao-VanJS) [inline addons/van_jsx/package.json:28]: Maybe add your name here?

File: addons/van_jsx/src/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { State, StateView } from "vanjs-core";
+
+export interface ReadonlyState<T> extends StateView<T> {
```
Comment (Tao-VanJS) [inline addons/van_jsx/src/index.d.ts:3]: ooc, do you need the interface `ReadonlyState<T>` here? I think `StateView<T>` already has the `readonly` `val` property.

**Review by Tao-VanJS — CHANGES_REQUESTED** [review]

**Comment by cqh963852** [comment]
@Tao-VanJS updated, PTAL

File: addons/van_jsx/src/index.d.ts (hunk 3 above)
Comment (cqh963852) [reply addons/van_jsx/src/index.d.ts:3]: should we rename createState to createRef ? but createRef is same as react legacy api createRef

**Comment by Tao-VanJS** [comment]
Thanks so much! @cqh963852

**Pull request merged**